core/src/main/java/com/linecorp/armeria/common/AggregatedHttpResponse.java

```diff
@@ -0,0 +1,247 @@
+        requireNonNull(status, "status");
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        return of(status, mediaType, HttpData.of(content, offset, length));
+    }
+
+    /**
+     * Creates a new HTTP response of the specified {@link HttpStatus}.
+     *
+     * @param mediaType the {@link MediaType} of the response content
+     * @param content the content of the response
+     */
+    static AggregatedHttpResponse of(HttpStatus status, MediaType mediaType, HttpData content) {
+        requireNonNull(status, "status");
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        return of(status, mediaType, content, HttpHeaders.of());
+    }
+
+    /**
+     * Creates a new HTTP response of the specified {@link HttpStatus}.
+     *
+     * @param mediaType the {@link MediaType} of the response content
+     * @param content the content of the response
+     * @param trailingHeaders the trailing HTTP headers
+     */
+    static AggregatedHttpResponse of(HttpStatus status, MediaType mediaType, HttpData content,
+                                     HttpHeaders trailingHeaders) {
+        requireNonNull(status, "status");
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        requireNonNull(trailingHeaders, "trailingHeaders");
+
+        final ResponseHeaders headers = ResponseHeaders.builder(status)
+                                                       .contentType(mediaType)
+                                                       .build();
+        return of(headers, content, trailingHeaders);
+    }
+
+    /**
+     * Creates a new HTTP response with empty content.
+     *
+     * @param headers the HTTP headers
+     */
+    static AggregatedHttpResponse of(ResponseHeaders headers) {
+        requireNonNull(headers, "headers");
+        return of(headers, HttpData.EMPTY_DATA, HttpHeaders.of());
+    }
+
+    /**
+     * Creates a new HTTP response.
+     *
+     * @param headers the HTTP headers
+     * @param content the content of the HTTP response
+     */
+    static AggregatedHttpResponse of(ResponseHeaders headers, HttpData content) {
+        requireNonNull(headers, "headers");
+        requireNonNull(content, "content");
+        return of(headers, content, HttpHeaders.of());
+    }
+
+    /**
+     * Creates a new HTTP response.
+     *
+     * @param headers the HTTP headers
+     * @param content the content of the HTTP response
+     * @param trailingHeaders the trailing HTTP headers
+     */
+    static AggregatedHttpResponse of(ResponseHeaders headers, HttpData content, HttpHeaders trailingHeaders) {
+        requireNonNull(headers, "headers");
+        requireNonNull(content, "content");
+        requireNonNull(trailingHeaders, "trailingHeaders");
+        return of(Collections.emptyList(), headers, content, trailingHeaders);
```

Comment:
We usually use `ImmutableList.of()` for empty list so guava can optimize for it sometimes.

Reply:
Ah just copied from the old source without much thinking. Thanks!